File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    int len = s.size();
    bool answer = true;
    int left = 0;
    int right = 0;
    
    for(int i = 0; i < len ; i ++){
        if(right > left) {
            answer = false;
            break;
        }
        if(s[i] == '(') left++;
        else right++;
    }
    if(left != right) answer = false;
    return answer;
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for now in s:
        if now == ')':
            if len(stack) != 0 and stack[-1] == '(': stack.pop()
            else: stack.append(now)
        else:
            stack.append(now)
    
    answer = True if len(stack) == 0 else False
    return answer